### PR TITLE
Backend tensorflow.compat.v1 supports forward-mode automatic differentiation via double backwards trick

### PR DESCRIPTION
**Pull request by ZongrenZou**

### CODE REVIEW

File: deepxde/gradients/gradients_forward.py [hunk 1]
```diff
@@ -20,19 +20,27 @@ def __call__(self, i=None, j=None):
 
         # Compute J[:, j]
         if j not in self.J:
-            if backend_name in [
-                "tensorflow.compat.v1",
-                "paddle",
-            ]:
+            if backend_name == "paddle":
```
Comment (lululxvi) [inline deepxde/gradients/gradients_forward.py:23]: Reorder as tf1, tf, pytorch, jax, paddle

File: deepxde/gradients/gradients_forward.py [hunk 2]
```diff
@@ -87,6 +95,8 @@ def grad_fn(x):
                     self.J[j][0][:, i : i + 1],
                     lambda x: self.J[j][1](x)[i : i + 1],
                 )
+            elif backend_name == "tensorflow.compat.v1":
```
Comment (lululxvi) [inline deepxde/gradients/gradients_forward.py:98]: Put tf1 as the first `if`

Comment (ZongrenZou) [reply deepxde/gradients/gradients_forward.py]: Done

File: deepxde/gradients/gradients_forward.py (hunk 1 above)
Comment (ZongrenZou) [reply deepxde/gradients/gradients_forward.py]: Done
